fhir-smart/src/main/java/com/ibm/fhir/smart/AuthzPolicyEnforcementPersistenceInterceptor.java

```diff
@@ -0,0 +1,268 @@
+        Map<ContextType, List<Scope>> approvedScopeMap = approvedScopes.stream()
+                // First filter the list to only scopes which grant the required permissions on the passed resourceType
+                .filter(s -> s.getResourceType() == ResourceType.ValueSet.RESOURCE ||
+                        s.getResourceType().value().equals(resourceType))
+                .filter(s -> hasPermission(s.getPermission(),requiredPermission))
+                // Then group the scopes by their context type
+                .collect(Collectors.groupingBy(s -> s.getContextType()));
+
+        if (approvedScopeMap.containsKey(ContextType.PATIENT)) {
+
+            // If the target resource is the Patient resource which matches the in-context patient, allow it
+            if (resource instanceof Patient && resource.getId() != null && contextIds.contains(resource.getId())) {
+                if (log.isLoggable(Level.FINE)) {
+                    log.fine("Interaction is permitted via scope " + approvedScopeMap.get(ContextType.PATIENT) +
+                        " with patient context '" + resource.getId() + "'");
+                }
+                return;
+            }
+
+            // Else, see if the target resource belongs to the Patient compartment of the in-context patient
+            try {
+                List<String> inclusionCriteria = CompartmentUtil
+                        .getCompartmentResourceTypeInclusionCriteria(CompartmentType.PATIENT.getValue(), resourceType);
+
+                EvaluationContext resourceContext = new FHIRPathEvaluator.EvaluationContext(resource);
+
+                for (String searchParm : inclusionCriteria) {
+                    try {
+                        String expression = SearchUtil.getSearchParameter(resourceType, searchParm).getExpression().getValue();
+                        Collection<FHIRPathNode> nodes = FHIRPathEvaluator.evaluator().evaluate(resourceContext, expression);
+                        for (FHIRPathNode node : nodes) {
+                            String patientRefVal = getPatientRefVal(node);
+                            if (patientRefVal != null && contextIds.contains(patientRefVal)) {
+                                if (log.isLoggable(Level.FINE)) {
+                                    log.fine("Interaction is permitted via scope " + approvedScopeMap.get(ContextType.PATIENT) +
+                                        " with patient context '" + patientRefVal + "'");
+                                }
+                                return;
+                            }
+                        }
+                    } catch (Exception e) {
+                        log.log(Level.WARNING, "Unexpected exception while processing inclusionCriteria '" + searchParm +
+                                "' in the Patient compartment for resource type " + resourceType);
+                    }
+                }
+            } catch (FHIRSearchException e) {
```

Comment:
FHIRSearchException... shouldn't this be an operation exception?

Reply:
this is whats thrown from `CompartmentUtil.getCompartmentResourceTypeInclusionCriteria`.  Potentially that could get refactored to throw something else, but I think its beyond the scope of this PR.  Please re-open if you disagree.

Reply:
this was more of a confirmation question.